x-pack/plugin/sql/qa/server/src/main/resources/version.csv-spec

```diff
@@ -0,0 +1,367 @@
+// To mute tests follow example in file: example.csv-spec
+
+//
+// Tests for VERSION fields
+//
+
+selectAll
+SELECT * FROM apps ORDER BY id;
+
+ id:i |  name:s   |  version:s
+------+-----------+---------
+1     |aaaaa      |1
+2     |bbbbb      |2.1
+3     |ccccc      |2.3.4
+4     |ddddd      |2.12.0
+5     |eeeee      |1.11.0
+6     |fffff      |5.2.9
+7     |ggggg      |5.2.9-SNAPSHOT
+8     |hhhhh      |1.2.3.4
+9     |iiiii      |bad
+10    |jjjjj      |5.2.9
+11    |kkkkk      |null
+12    |aaaaa      |1.2.3.4
+13    |lllll      |null
+14    |mmmmm      |5.2.9
+
+;
+
+filterByVersion
+SELECT * FROM apps WHERE version = '2.12.0';
+
+  id  |  name     |  version
+------+-----------+---------
+4     |ddddd      |2.12.0
+;
+
+keywordSubfield
+SELECT version.raw as k FROM apps WHERE version = '2.12.0';
+
+  k
+---------
+ 2.12.0
+;
+
+
+projectionVersion
+SELECT version FROM apps WHERE id = 3;
+
+ version
+---------
+2.3.4
+;
+
+
+versionRange1
+SELECT * FROM apps WHERE version > '2.2' ORDER BY id;
+
+  id  |  name     |  version
+------+-----------+---------
+3     |ccccc      |2.3.4
+4     |ddddd      |2.12.0
+6     |fffff      |5.2.9
+7     |ggggg      |5.2.9-SNAPSHOT
+9     |iiiii      |bad
+10    |jjjjj      |5.2.9
+14    |mmmmm      |5.2.9
+
+;
+
+versionRange2
+SELECT * FROM apps WHERE version >= '2.3.4' ORDER BY id;
+
+  id  |  name     |  version
+------+-----------+---------------
+3     |ccccc      |2.3.4
+4     |ddddd      |2.12.0
+6     |fffff      |5.2.9
+7     |ggggg      |5.2.9-SNAPSHOT
+9     |iiiii      |bad
+10    |jjjjj      |5.2.9
+14    |mmmmm      |5.2.9
+
+;
+
+
+between
+SELECT * FROM apps WHERE version BETWEEN '1.10' AND '5.2.9' ORDER BY id;
+
+  id:i  |  name:s     |  version:s
+--------+-------------+---------------
+2       |bbbbb        | 2.1
+3       |ccccc        | 2.3.4
+4       |ddddd        | 2.12.0
+5       |eeeee        | 1.11.0
+6       |fffff        | 5.2.9
+7       |ggggg        | 5.2.9-SNAPSHOT
+10      |jjjjj        | 5.2.9
+14      |mmmmm        | 5.2.9
+
+;
+
+rlike
+SELECT * FROM apps WHERE version::string RLIKE '2.*.4';
+
+  id:i  |  name:s     |  version:s
+--------+-------------+---------------
+3       |ccccc        | 2.3.4
+
+;
+
+orderByVersion
+SELECT * FROM apps ORDER BY version, id;
```

Comment:
At first I thought it could depend on valid vs. invalid version numbers, but I tried it on different datasets and it seems to be completely independent from that.
I investigated it a bit more and it seems it depends on the same problem reported in #58820